### PR TITLE
test/storagelib: step up our destroy-and-reboot efforts

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Instead of trying to gracefully shutdown a destroyed system from the inside, do it forcefully from the outside.

### CODE REVIEW

File: test/common/storagelib.py [hunk 1]
```diff
@@ -554,8 +554,13 @@ def encrypt_root(self, passphrase):
 grubby --update-kernel=ALL --args="root=UUID=$uuid rootflags=defaults rd.luks.uuid=$luks_uuid rd.lvm.lv=root/root"
 ! test -f /etc/kernel/cmdline || cp /etc/kernel/cmdline /new-root/etc/kernel/cmdline
 """, timeout=300)
-        m.spawn("dd if=/dev/zero of=/dev/vda bs=1M count=100; reboot", "reboot", check=False)
-        m.wait_reboot(300)
+
+        # Destroy our current boot device.  This prevents the system from
+        # shutting down properly, so do it forcefully from outside.
+        m.spawn("sync; dd if=/dev/zero of=/dev/vda bs=1M count=100; sync; sync; sync;")
```
Comment (martinpitt) [inline test/common/storagelib.py:560]: You want m.execute() for that to make it sync. Will also fix [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-20449-3ac228be-20240508-064614-rhel-8-10-distropkg-storage/log.html#88)

**Comment by martinpitt** [comment]
But more generally, this is just cosmetics -- it's *not* the shutdown that fails, it's the boot. The systemd askpass agent doesn't properly hand over the passphrase to stratis in the initrd. See https://github.com/cockpit-project/cockpit/pull/20373#issuecomment-2081939381

**Comment by martinpitt** [comment]
@allisonkarlitskaya This is still buggy and wasn't needed. Do you still want to work on this, or should we just close?

**Comment by allisonkarlitskaya** [comment]
Ya.  This was a false understanding on my part about the core of the problem.  kann weg.

**Pull request closed without merging**